precompiles/xcm/Xtokens.sol

```diff
@@ -0,0 +1,119 @@
+        WeightV2 memory weight
+    ) external returns (bool);
+
+    /// Transfer a token through XCM based on its currencyId specifying fee
+    ///
+    /// @dev The token transfer burns/transfers the corresponding amount before sending
+    /// @param currencyAddress The ERC20 address of the currency we want to transfer
+    /// @param amount The amount of tokens we want to transfer
+    /// @param destination The Multilocation to which we want to send the tokens
+    /// @param weight The weight we want to buy in the destination chain 
+    function transfer_with_fee(
+        address currencyAddress,
+        uint256 amount,
+        uint256 fee,
+        Multilocation memory destination,
+        WeightV2 memory weight
+    ) external returns (bool);
+
+    /// Transfer a token through XCM based on its MultiLocation
+    ///
+    /// @dev The token transfer burns/transfers the corresponding amount before sending
+    /// @param asset The asset we want to transfer, defined by its multilocation.
+    /// Currently only Concrete Fungible assets
+    /// @param amount The amount of tokens we want to transfer
+    /// @param destination The Multilocation to which we want to send the tokens
+    /// @param weight The weight we want to buy in the destination chain 
+    function transfer_multiasset(
+        Multilocation memory asset,
+        uint256 amount,
+        Multilocation memory destination,
+        WeightV2 memory weight
+    ) external returns (bool);
+
+    /// Transfer a token through XCM based on its MultiLocation specifying fee
+    ///
+    /// @dev The token transfer burns/transfers the corresponding amount before sending
+    /// @param asset The asset we want to transfer, defined by its multilocation.
+    /// Currently only Concrete Fungible assets
+    /// @param amount The amount of tokens we want to transfer
+    /// @param destination The Multilocation to which we want to send the tokens
+    /// @param weight The weight we want to buy in the destination chain 
+    function transfer_multiasset_with_fee(
+        Multilocation memory asset,
+        uint256 amount,
+        uint256 fee,
+        Multilocation memory destination,
+        WeightV2 memory weight
+    ) external returns (bool);
+
+    /// Transfer several tokens at once through XCM based on its address specifying fee
+    ///
+    /// @dev The token transfer burns/transfers the corresponding amount before sending
+    /// @param currencies The currencies we want to transfer, defined by their address and amount.
+    /// @param feeItem Which of the currencies to be used as fee
+    /// @param destination The Multilocation to which we want to send the tokens
+    /// @param weight The weight we want to buy in the destination chain 
+    function transfer_multi_currencies(
+        Currency[] memory currencies,
+        uint32 feeItem,
+        Multilocation memory destination,
+        WeightV2 memory weight
+    ) external returns (bool);
```

Comment:
Do we support this in our runtimes?

Reply:
Any particular reason for this question?
In our configs, we support 2 assets for transfer.
`pub const MaxAssetsForTransfer: usize = 2;`

Reply:
I meant the `Currency` type actually.

I thought we omitted support for it, but we're just using numeric asset id. You can ignore this comment.